Spec/RealtimeClientConnection.swift

```diff
@@ -2103,36 +2103,59 @@ class RealtimeClientConnection: QuickSpec {
+                        options.logLevel = .Verbose
                         options.disconnectedRetryTimeout = 1.0
+                        options.tokenDetails = getTestTokenDetails(ttl: 2.0)
                         let client = AblyTests.newRealtime(options)
                         defer { client.dispose(); client.close() }
-                        let channel = client.channels.get("test")
-
-                        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
 
-                        let expectedConnectionId = client.connection.id
+                        let channel = client.channels.get("test")
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.attach() { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                        }
 
                         guard let transport = client.transport as? TestProxyTransport else {
                             fail("TestProxyTransport is not set"); return
                         }
 
-                        let disconnectedMessage = ARTProtocolMessage()
-                        disconnectedMessage.action = .Disconnected
-                        disconnectedMessage.connectionId = "xxxxxxxxx"
-                        transport.receive(disconnectedMessage)
+                        let initialConnectionId = client.connection.id
+
+                        // Wait for token to expire
+                        expect(transport.protocolMessagesReceived.filter{ $0.action == .Disconnected }).toEventually(haveCount(1), timeout: testTimeout)
 
                         waitUntil(timeout: testTimeout) { done in
-                            // Resumed
+                            let partialDone = AblyTests.splitDone(2, done: done)
+                            channel.once(.Detached) { error in
+                                expect(error).to(beNil())
+                                partialDone()
+                            }
+                            // Wait for connection resume
                             client.connection.once(.Connected) { stateChange in
-                                guard let stateChange = stateChange else {
-                                    fail("ConnectionStateChange is nil"); done(); return
-                                }
-                                expect(client.connection.id).to(equal(disconnectedMessage.connectionId))
-                                done()
+                                expect(stateChange?.reason).to(beNil())
+                                partialDone()
                             }
                         }
 
-                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
+                        expect(client.connection.id).toNot(equal(initialConnectionId))
```

Comment:
From the logs, when the token expires the `connectionId` always changes, so I decided to add that and additionally it checks if the channel gets detached.